Public/Sdk/SelfHost/BuildXL/Tools/QTest/Tool.QTestRunner.dsc

```diff
@@ -300,6 +340,10 @@ export interface QTestArguments extends Transformer.RunnerArguments {
     weight?: number;
     /** Privilege level required by this process to execute. */
     privilegeLevel?: "standard" | "admin";
+     /** Directory to write coverage logs in */
+    coverageLogsDir?: Directory;
+    /** Specifies whether this is a debug or retail build */
+    qTestBuildType?: string;

```

Comment:
DScript uses the concept of qualifiers to track this information. Qualifier types are user defined based on build specs though, so in this SDK it may not be straightforward to make sure the types are compatible.

So instead of using a string, or the qualifier functionality, I'd suggest you emulate what was done with qTestPlatform so the argument is still strongly typed

Reply:
So qTestBuildType is really a string type: it can accept any valid string value. I'm not sure I understand what you mean when you say "so the argument is still strongly typed"

Reply:
I was suggesting you define it as an enum instead of a string

`
@@public
export const enum QTestBuildType {
    debug,
    release,
}`

Reply:
Please use union strings instead of Enums, they are more extensible, easier to read and more future proof. The type safetly of union of string literal is identical to enum.s

Reply:
`type QTstBuildType = "debug" | "release";`

Reply:
Is this really resolved?  The type is still `string`.

Reply:
I'm not fixing this: the type that qtest expects is really string - all string values (not just "debug"/"retail") are valid